README.md

```diff
@@ -3,38 +3,63 @@ The riff Buildpack is a Cloud Native Buildpack V3 that provides riff Invokers to
 
+## Buildpack Behavior
 
-## Detection Phase
+### Detection Phase
 Detection passes if 
 - a `$APPLICATION_ROOT/riff.toml` exists and 
 - Either
-  1. the build plan already contains a `jvm-application` key
-  2. as a fallback, the file pointed by the `artifact` value in `riff.toml` exists and is executable
+    1. the build plan already contains a `jvm-application` key (typically because a JVM based application was detected by the [java buildpack](https://github.com/cloudfoundry/build-system-buildpack))
+    2. the build plan already contains a `npm` key (typically because an NPM based application was detected by the [npm buildpack](https://github.com/cloudfoundry/npm-cnb))
+        
+        1. alternatively, if the file pointed by the artifact value in `riff.toml` exists and has a `.js` extension
+    3. as a fallback, the file pointed by the `artifact` value in `riff.toml` exists and is executable
     
 If detection passes in (i), the buildpack will contribute an `openjdk-jre` key with `launch` metadata to instruct 
 the `openjdk-buildpack` to provide a JRE.  It will also add a `riff-invoker-java` key and `handler` 
 metadata extracted from the riff metadata.
 
-If detection passes in (ii), the buildpack will add a `riff-invoker-command` key and `command` 
+If detection passes in (ii), the buildpack will add a `riff-invoker-node` key and `fn` 
 metadata extracted from the riff metadata.
 
-## Build Phase
+If detection passes in (iii), the buildpack will add a `riff-invoker-command` key and `command` 
+metadata extracted from the riff metadata.
+
+If several languages are detected simultaneously, the detect phase errors out.
+The `override` key in `riff.toml` can be used to bypass detection and force the use of a particular invoker.
+
+### Build Phase
 
 If a java build has been detected
-* Contributes riff Java Invoker to a launch layer, set as the main java entry point with `function.uri = <build>?handler=<handler>` set
+* Contributes the riff Java Invoker to a launch layer, set as the main java entry point with `function.uri = <build-directory>?handler=<handler>` set as an environment variable.
+
+If a node function has been detected
+* Contributes the riff Node Invoker to a launch layer, set as the main `node` entry point with `FUNCTION_URI = <artifact>` set as an environment variable.
+Note that `artifact` may actually be empty, in which case the invoker will `require()` the current directory (the function), which in turn expects that it contains a valid `package.json` file with its `main` entry point set. 
 
 If a command function has been selected
-* Contributes the riff Command Invoker to a launch layer, set as the main executable with `FUNCTION_URI = <artifact>`
+* Contributes the riff Command Invoker to a launch layer, set as the main executable with `FUNCTION_URI = <artifact>` set as an environment variable.
 
 
 * Contributes `web` process
 * Contributes `function` process
 
+## In Plain English
```

Comment:
i would suggest putting this section near the top, and and removing the first sentence below.